ragas_eval.py

```
@@ -0,0 +1,211 @@
+        {
+            "model": model,
+            "messages": [{"role": "user", "content": question}],
+            "temperature": temperature,
+        }
+    ).encode()
+
+    req = urllib.request.Request(
+        f"{url}/v1/chat/completions",
+        data=payload,
+        headers=headers,
+    )
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)